src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -220,67 +248,68 @@ private void addDeclaredDependencies(@NotNull ErlangFile erlangModule, @NotNull
       Set<String> behaviours = ContainerUtil.newHashSet();
       ErlangPsiImplUtil.addDeclaredBehaviourModuleNames(erlangModule, behaviours);
-      return getPathsFromNames(behaviours);
+      return resolvePathsFromNames(behaviours);
     }
 
     @NotNull
-    private List<String> getDeclaredParseTransformPaths(@NotNull ErlangFile erlangModule) {
-      Set<String> pt = ContainerUtil.newHashSet();
-      erlangModule.addDeclaredParseTransforms(pt);
-      return getPathsFromNames(pt);
-    }
-
-    private void buildNamesMap(@NotNull Collection<VirtualFile> erlangModules) {
-      for (VirtualFile erlangModule : erlangModules) {
-        myNamesToPathsMap.put(erlangModule.getNameWithoutExtension(), erlangModule.getPath());
+    private List<String> resolvePathsFromNames(@NotNull Collection<String> erlangModuleNames) {
+      List<String> paths = ContainerUtil.newArrayList();
+      for (String erlangModuleName : erlangModuleNames) {
+        paths.addAll(getPathsFromModuleName(erlangModuleName));
       }
+      return paths;
     }
 
     @NotNull
-    private List<String> getPathsFromNames(@NotNull Collection<String> erlangModuleNames) {
-      return ContainerUtil.mapNotNull(erlangModuleNames, new Function<String, String>() {
+    private List<String> getDeclaredIncludePaths(@NotNull ErlangFile file) {
+      return ContainerUtil.mapNotNull(ErlangPsiImplUtil.getDirectlyIncludedFiles(file), new Function<ErlangFile, String>() {
+        @Nullable
         @Override
-        public String fun(String name) {
-          return myNamesToPathsMap.get(name);
+        public String fun(ErlangFile erlangFile) {
+          VirtualFile virtualFile = erlangFile.getVirtualFile();
+          return virtualFile != null && myPathsToDependenciesMap.containsKey(virtualFile.getPath()) ? virtualFile.getPath() : null;
         }
       });
     }
 
-    @Override
-    public Collection<String> getNodes() {
-      return myPathsToDependenciesMap.keySet();
+    @NotNull
+    private List<String> getPathsFromModuleName(@NotNull String erlangModuleName) {
+      return ContainerUtil.mapNotNull(ErlangModuleIndex.getFilesByName(myProject, erlangModuleName, GlobalSearchScope.projectScope(myProject)), new Function<ErlangFile, String>() {
+        @Nullable
+        @Override
+        public String fun(ErlangFile erlangFile) {
+          VirtualFile virtualFile = erlangFile.getVirtualFile();
+          return virtualFile != null ? virtualFile.getPath() : null;
+        }
+      });
     }
+  }
 
-    @Override
-    public Iterator<String> getIn(String filePath) {
-      return myPathsToDependenciesMap.get(filePath).iterator();
+  static class CyclicDependencyFoundException extends Exception {
+    private final Couple<String> myCircularDependencies;
+
+    CyclicDependencyFoundException(@NotNull Couple<String> mytCircularDependencies) {
+      this.myCircularDependencies = mytCircularDependencies;
```

Comment:
There is a difference between "cyclic" and "circular" - even in Russian. Method parameter is named improperly, and it has a typo, as well.